osgp/platform/osgp-adapter-ws-core/src/main/java/org/opensmartgridplatform/adapter/ws/core/application/services/DeviceManagementService.java

```diff
@@ -233,6 +235,25 @@ public Organisation findOrganisation(@Identification final String organisationId
         return this.eventRepository.findAll(specification, request);
     }
 
+    private Specification<Event> handleDescription(final String description, final String descriptionStartsWith,
+            Specification<Event> specification) throws ArgumentNullOrEmptyException {
+        if (description == null) {
```

Comment:
like this (also see the closed source JpaDeviceSpecifications class for similar code), in case the "or" remains:
Specification<Event> descriptionSpecification = this.eventSpecifications(...);
Specification<Event> descriptionStartsWithSpecification = this.eventSpecifications(...);

return specification.and(descriptionSpecification.or(descriptionStartsWithSpecification));

And in the eventSpecifications methods:
if (description == null) {
 return (eventRoot, query, cb) -> cb.and();
}

return ((eventRoot, query, cb) -> cb.equal(eventRoot.<String> get(DESCRIPTION), description));

When using the "cb.and()" construction, you can also remove the "throws ArgumentNullOrEmptyException" clause.

Reply:
tried this
when filtering only on Description or only on DescriptionStartsWith i did not get the correct results.
Therefor I had to use a different construction